src/main/java/io/confluent/idesidecar/restapi/kafkarest/controllers/TopicManagerImpl.java

```diff
@@ -0,0 +1,156 @@
+import io.vertx.core.http.HttpServerRequest;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import org.apache.kafka.clients.admin.DescribeTopicsOptions;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicDescription;
+
+/**
+ * RequestScoped bean for managing Kafka topics. Creating the bean as {@link RequestScoped} allows
+ * us to inject the {@link HttpServerRequest} which is used to get the connection ID from the
+ * request headers.
+ */
+@RequestScoped
+public class TopicManagerImpl implements TopicManager {
+
+  @Inject
+  AdminClients adminClients;
+
+  @Inject
+  ClusterManagerImpl clusterManager;
+
+  @Inject
+  HttpServerRequest request;
+
+  Supplier<String> connectionId = () -> request.getHeader(CONNECTION_ID_HEADER);
+
+  @Override
+  public Uni<TopicData> createKafkaTopic(String clusterId,
+      CreateTopicRequestData createTopicRequestData) {
+    return clusterManager.getKafkaCluster(clusterId)
+        .chain(ignored -> uniStage(
+            adminClients
+                .getAdminClient(connectionId.get(), clusterId)
+                .createTopics(List.of(new NewTopic(
+                    createTopicRequestData.getTopicName(),
+                    Optional.ofNullable(createTopicRequestData.getPartitionsCount())
+                        .orElse(1),
+                    Optional.ofNullable(createTopicRequestData.getReplicationFactor())
+                        .orElse(1).shortValue())
+                )).all().toCompletionStage()))
+        .chain(v -> getKafkaTopic(
+            clusterId,
+            createTopicRequestData.getTopicName(),
+            false
+        ));
+  }
+
+  @Override
+  public Uni<Void> deleteKafkaTopic(String clusterId, String topicName) {
+    return clusterManager.getKafkaCluster(clusterId).chain(ignored ->
+        uniStage(
+            adminClients.getAdminClient(connectionId.get(), clusterId)
+                .deleteTopics(List.of(topicName))
+                .all()
+                .toCompletionStage())
+    );
+  }
+
+  @Override
+  public Uni<TopicData> getKafkaTopic(
+      String clusterId, String topicName, Boolean includeAuthorizedOperations
+  ) {
+    var describeTopicsOptions = new DescribeTopicsOptions()
+        .includeAuthorizedOperations(
+            Optional.ofNullable(includeAuthorizedOperations).orElse(false)
+        );
+    return clusterManager.getKafkaCluster(clusterId).chain(ignored -> uniStage(
+        adminClients.getAdminClient(connectionId.get(), clusterId)
+            .describeTopics(List.of(topicName), describeTopicsOptions)
+            .allTopicNames()
+            .toCompletionStage()
+    )
+        .map(topicDescriptions -> topicDescriptions.values().iterator().next())
+        .onItem().transform(topicDescription -> fromTopicDescription(clusterId, topicDescription)));
```

Comment:
nit: This is hard to read. What about something like:

```suggestion
    return clusterManager
        .getKafkaCluster(clusterId)
        .chain(ignored ->
            uniStage(
              adminClients
                  .getAdminClient(connectionId.get(), clusterId)
                  .describeTopics(List.of(topicName), describeTopicsOptions)
                  .allTopicNames()
                  .toCompletionStage()
            )
            .map(topicDescriptions -> topicDescriptions.values().iterator().next())
            .onItem()
            .transform(topicDescription -> fromTopicDescription(clusterId, topicDescription))
        );
```